src/Aspire.Hosting/Dashboard/DashboardLifecycleHook.cs

```diff
@@ -0,0 +1,147 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Aspire.Dashboard.Model;
+using Aspire.Hosting.ApplicationModel;
+using Aspire.Hosting.Dcp;
+using Aspire.Hosting.Lifecycle;
+using Aspire.Hosting.Utils;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+
+namespace Aspire.Hosting.Dashboard;
+
+internal sealed class DashboardLifecycleHook(IConfiguration configuration,
+                                             IOptions<DcpOptions> dcpOptions,
+                                             ILogger<DistributedApplication> distributedApplicationLogger,
+                                             IDashboardEndpointProvider dashboardEndpointProvider) : IDistributedApplicationLifecycleHook
+{
+    public Task BeforeStartAsync(DistributedApplicationModel model, CancellationToken cancellationToken)
```

Comment:
For bonus points, you could double check that the host is in run mode and dashboard isn't disabled here. They should always be true for this hook to be added, but this could be a double check against a bad change in the future.

Reply:
I can make it an asset